diff -uNr ClibPDF.orig/source/config.h ClibPDF/source/config.h
--- ClibPDF.orig/source/config.h	Fri Jan 28 04:26:01 2000
+++ ClibPDF/source/config.h	Sun Jul 21 00:47:13 2002
@@ -196,16 +196,16 @@
 /* Acrobat Reader 3.0x for Linux works beautifully. */
 #ifdef Linux
 #define PLATFORM_NAME			"Linux"
-#define PDF_VIEWER_COMMAND		"/usr/local/Acrobat4/bin/acroread"
+#define PDF_VIEWER_COMMAND		"xpdf"
 #define TEMP_DIRECTORY			"/tmp/"
-#define ZLIB_COMPRESS_PATH		"/usr/local/bin/zlibcomp"
+#define ZLIB_COMPRESS_PATH		"" /* not needed zlib is dispo (HGO) */
 #define BINARY_WRITE			"w"
 #define BINARY_READ			"r"
 #define DIR_SEPARATOR			'/'
 #define UNIX				1
 #define DEF_FONTMAPFILE			"fontmap.lst"
-#define DEF_PFM_DIR			"/usr/share/fonts/default/Type1"
-#define DEF_PFB_DIR			"/usr/share/fonts/default/Type1"
+#define DEF_PFM_DIR			"/usr/share/fonts/Type1/pfm"
+#define DEF_PFB_DIR			"/usr/share/fonts/Type1"
 #endif
 
 /* MacOS X (Server) is Unix */
